digdag-standards/src/main/java/io/digdag/standards/command/ecs/EcsClientConfig.java

```diff
@@ -33,29 +33,31 @@ public EcsClientConfig(EcsClientConfigBuilder builder)
         this.assignPublicIp = builder.isAssignPublicIp();
     }
 
-    public static EcsClientConfig createFromTaskConfig(final Optional<String> clusterName, final Config config)
+    public static EcsClientConfig createFromTaskConfig(final Optional<String> clusterName, final Config taskConfig, final Config systemConfig)
     {
         final String name;
-        // `config` is assumed to have a nested config with following values
-        // at the key of `TASK_CONFIG_ECS_KEY` from `config`.
+        // `taskConfig` is assumed to have a nested taskConfig with following values
+        // at the key of `TASK_CONFIG_ECS_KEY` from `taskConfig`.
         // - launch_type (optional)
-        // - access_key_id
```

Comment:
Where were these fields set in the nested config?

Reply:
I expect that these fields are set in the caller (e.g. `CommandExecutor` classes)

Reply:
Oh, I just thought we might need to remove these system credentials (access_key_id and secret_access_key) from all task configs as well in this PR., but you didn't add any code that puts the credentials in a task config? If so, I'm not sure these system credentials are set by callers...

Reply:
Ah sorry. Maybe I wrongly understood what you meant. 

I thought that you have mentioned about **ALL** fields of them ( including `launch_type`, `region`, `subnets` and so on). But it seems that you mentioned about **ONLY** `access_key_id` and `secret_access_key`, is it correct?

With explaining the background from the beginning,

1. In the first time when `EcsCommandExecutor` was released, `access_key_id` and `secret_access_key` were expected to be put only into `systemConfig`. So `EcsClientConfig` had been built only from `systemConfig`.
1. At https://github.com/treasure-data/digdag/pull/1469, I added a support of `building EcsClientConfig from taskConfig` so that we can change the contents of `EcsClientConfig` for each `taskConfig`.
1. After merging #1469, I thought that sensitive data should not be included in `taskConfig`. So I sent this PR so that `access_key_id` and `secret_access_key` are retrieved only from `systemConfig`.

> Oh, I just thought we might need to remove these system credentials (access_key_id and secret_access_key) from all task configs as well in this PR.

Yes, you are correct. 

>  but you didn't add any code that puts the credentials in a task config?  

Implicitly it depended on users whether `access_key_id` and `secret_access_key` were put into the taskConfig not. Surely it was not good implementation. But this PR fixes it as well.

So this PR does not include any code change to remove system credentials from task configs. 

In the original implementation, `access_key_id` and `secret_access_key` were expected from `server.properties` and can be fetched with `agent.command_executor.ecs.{cluster_name}.access_key_id` and `agent.command_executor.ecs.{cluster_name}.secret_access_key` repeatedly as follows.
https://github.com/treasure-data/digdag/blob/ead8c27608b6e01a644a794c1910d6b4f50a63cb/digdag-standards/src/main/java/io/digdag/standards/command/ecs/EcsClientConfig.java#L48-L57

>  If so, I'm not sure these system credentials are set by callers...

As I explained above, they were expected to be set `systemConfig` implicitly. So originally there has not been the explicit implementations to do that.